### DISCUSSION TITLE
Alternative to injectfilter / inject js in html

### DISCUSSION THREAD

**Discussion by joaocosteira** [vikejs/vike#1062] [Discussion]
Hello, as you may already know for previous discussions, I've been using vite-plugin-ssr and by combining the programmatic build/prerender functionality, I'm able to ssg whatever I want.

Now the issue that I'm having is being able to inject the js that was generated within the html itself, instead of having all of those module/preload tags in the html. Ideally I would love to have a script tag with all of the necessary js within it.
 
As I already tested and based on other discussions found on this git, there's the vite-plugin-singlefile, but configuring it or simply trying to disable chunking/setting up vite/rollup is being quite difficult with this plugin.

After reading the documentation and finding the  `injectfilter` what I'm wondering is if it is possible to use something similar in order to actually inject the js in the html or some kind of hint on how to configure the build/rollup. Because this plugin is great but is slightly confusing in some areas, because we don't have the classic index.html entry point and I'm kinda lost on what's the best approach: use vite library mode, customize the rollup options, write a plugin, what entry points to use etc.
 I know that I'm rambling, but I want some kind of hint/what's the best approach to achieve what I want.

Anyways, thank you, both of this plugin and the help provided in this git are amazing.

**Comment by brillout** [comment]
The ultimate goal is increased performance, correct?

**Reply by joaocosteira** [reply]
Not only that, but an interesting case that I'm trying to overcome is being able to see the results without running a server, by opening the html files generated by the ssg using the file protocol (js files are blocked via cors). Obviously we can easily see the results by running vite preview or installing server like it is recommended in the docs.

Having a folder with some js chunks wouldn't be the end of the world, but being able to double click on an html file and see the result without running a server would be very useful. That's why I'm trying to mess around with the build process in order to not use preloading or trying to merge everything within a single html file.

**Reply by brillout** [reply]
The thing is that merging everything into a single HTML file is a no-go from a caching perspective. (CC @spacedawwwg.)

Your second use case seems to be very niche. I ain't categorically against it but, as you can imagine, there's a lot of higher prio things going on.

**Reply by brillout** [reply]
But let me know if it somehow becomes a blocker.

**Reply by joaocosteira** [reply]
Sure, I know that this is very low priority, and I'm not expecting you guys to create a function that magically does everything, but can you guys provide some kind of lead on what's the best approach to try to figure this thing out, for instance I'm having some trouble trying to manipulate the js chuncks being generated, because like I already saw in other discussions, there's some confusion on how to customize the build process, for instance should we use the renderer folder as an entry point? like I said simply try to configure it using the vite build options or the underlying rollup config?

Anyways, thank you, I'll try to figure this thing out and in case I get some issue I'll get in touch, but any lead on what's the best approach to customize the build/prerender process would be very appreciated.

**Reply by brillout** [reply]
I understand, alas groking what appears to be a very niche use case, and thinking through a possible workaround for something that isn't a blocker is also time consuming. Which may be better spent on something else. If your company would sponsor, then that'd change the equation.

Yes, keep us updated 👍.

**Reply by brillout** [reply]
I don't think this will help you but FYI: https://github.com/brillout/vite-plugin-ssr/issues/695.
